src/MongoDB.Driver.Core/Core/Authentication/HashAlgorithmName.cs

```diff
@@ -0,0 +1,154 @@
+﻿/* Original work:
```

Comment:
From the CoreFX project.

Reply:
Removed readonly modifier from struct definition since our project does not use the version of C# where that is supported.
Added <inheritdoc/> comments.